plugins/view-resources/src/components/ActionHandler.svelte

```diff
@@ -165,73 +168,72 @@
       }
     }
 
-    // For none we ignore all actions.
-    if (ctx?.mode === 'none') {
-      return
-    }
     clearTimeout(timer)
 
-    currentActions = currentActions.filter(({ keyBinding, allowedForEditableContent }) => {
-      const hasKeyBinding = keyBinding !== undefined && keyBinding.length > 0
-      const allowed = !isContentEditable || allowedForEditableContent
-
-      return hasKeyBinding && allowed
-    })
-
-    if (lastKey !== undefined) {
-      for (const a of sequences) {
-        // TODO: Handle multiple keys here
-        if (a.keyBinding?.find((it) => (lastKey ? matchKeySequence(evt, it, lastKey) : false)) !== undefined) {
-          const action = await getResource(a.action)
-          if (action !== undefined) {
-            sequences = []
-            lastKey = undefined
-            delayedAction = undefined
-            Analytics.handleEvent(a._id)
-            await action(selectionDocs, evt, a.actionProps)
-            return
-          }
-        }
-      }
+    async function activateAction (a: Action): Promise<boolean> {
+      const action = await getResource(a.action)
+      if (action === undefined) return false
+
+      lastKey = undefined
+      delayedAction = undefined
+      Analytics.handleEvent(a._id)
+      const actionProps = { ...a.actionProps }
+      if (!Object.prototype.hasOwnProperty.call(actionProps, 'props')) actionProps.props = {}
+      actionProps.props.space = currentSpace
+      await action(selectionDocs, evt, actionProps)
+      return true
     }
 
-    sequences = getSequences(evt, currentActions)
-    let found = false
+    // 3 cases for keyBinding
+    //  matches the sequence - immediately action
+    //  start sequence - postpone other action
+    //  matches the key - execute if there is no start sequence actions
+
+    let postpone = false
+    let nonSequenceAction: Action | undefined
+
     for (const a of currentActions) {
-      // TODO: Handle multiple keys here
-      if (a.keyBinding?.find((it) => matchKey(evt, it)) !== undefined) {
-        const action = await getResource(a.action)
-        if (action !== undefined) {
-          if (sequences.length === 0) {
-            lastKey = undefined
-            sequences = []
-            delayedAction = undefined
-            Analytics.handleEvent(a._id)
-            await action(selectionDocs, evt, a.actionProps)
-            return
-          } else {
-            delayedAction = async () => {
-              Analytics.handleEvent(a._id)
-              await action(selectionDocs, evt, a.actionProps)
-            }
-            found = true
-          }
-        }
+      if (a.keyBinding === undefined || a.keyBinding.length < 1) continue
+      if (isContentEditable && (a.allowedForEditableContent !== true)) continue
+      const t = lastKey
+      if (t !== undefined &&
+        a.keyBinding.find((it) => matchKeySequence(evt, it, t)) !== undefined &&
```

Comment:
can we replace all `xx.find(...) !== undefined`  with `xx.some(...)` for clarity?